mdtraj/geometry/rdf.py

```diff
@@ -158,21 +160,30 @@ def compute_rdf_t(traj, pairs, times, period_length=None, r_range=None,
         pairs_set = np.unique(pairs)
         pairs = np.vstack([np.vstack([pairs_set, pairs_set]).T, pairs])
 
-    g_r_t = np.zeros(shape=(len(times), n_bins))
-    num_chunks = int(np.floor(traj.n_frames / period_length))
-
-    # Returns shape (len(times), len(pairs))
-    frame_distances = compute_distances_t(traj, pairs, times, periodic=periodic, opt=opt)
-
-    for n, distances in enumerate(frame_distances):
-        tmp, edges = np.histogram(distances, range=r_range, bins=n_bins)
-        g_r_t[n, :] += tmp
-    r = 0.5 * (edges[1:] + edges[:-1])
-
-    # Normalize by volume of the spherical shell (see above)
-    V = (4 / 3) * np.pi * (np.power(edges[1:], 3) - np.power(edges[:-1], 3))
-    norm = len(pairs) / (period_length) * np.sum(1.0 / traj.unitcell_volumes) * V
-
-    g_r_t = g_r_t.astype(np.float64) / norm  # From int64.
-
-    return r, g_r_t
+    g_r_t = []
+    weights = []
+
+    # Splits pairs into smaller chunks so that frame_distances is not excessively large
+    for i in range(np.ceil(len(pairs)/num_concurrent_pairs).astype("int")):
+        temp_g_r_t = np.zeros(shape=(len(times), n_bins))
+        pairs_set = pairs[i*num_concurrent_pairs:(i+1)*num_concurrent_pairs]
+        weights.append(len(pairs_set)/num_concurrent_pairs)
+        
+        # Returns shape (len(times), len(pairs_set))
+        frame_distances = compute_distances_t(traj, pairs_set, times, periodic=periodic, opt=opt)
+
+        for n, distances in enumerate(frame_distances):
+            tmp, edges = np.histogram(distances, range=r_range, bins=n_bins)
+            temp_g_r_t[n, :] += tmp
+        r = 0.5 * (edges[1:] + edges[:-1])
+
+        # Normalize by volume of the spherical shell (see above)
+        V = (4 / 3) * np.pi * (np.power(edges[1:], 3) - np.power(edges[:-1], 3))
+        norm = len(pairs_set) / (period_length) * np.sum(1.0 / traj.unitcell_volumes) * V
+
+        temp_g_r_t = temp_g_r_t.astype(np.float64) / norm  # From int64.
+        g_r_t.append(temp_g_r_t)
+
+    g_r_t_final = np.average(g_r_t, axis=0, weights=weights)
```

Comment:
Nice find - my worry with slicing it up like this was that the end "chunks" could be weighted weirdly if they were of strange length. Looks like this accounts for that.